src/libAtomVM/externalterm.c

```diff
@@ -217,6 +219,32 @@ static int serialize_term(Context *ctx, uint8_t *buf, term t)
         }
         return 5;
 
+#ifndef AVM_NO_FP
+    } else if (term_is_float(t)) {
+        if (!IS_NULL_PTR(buf)) {
+            avm_float_t val = term_to_float(t);
+            // ensure interoperable encoding of -0.0
+            if (val == -0.0) {
+                val = 0.0;
+            }
+#ifdef AVM_USE_SINGLE_PRECISION
+            buf[0] = FLOAT_EXT;
+            sprintf(buf + 1, "%.20e", val);
```

Comment:
This probably doesn't explain the armel crash, but we are leaking data here. Buf is not zeroed before we printf 25 or 26 characters (depending on the sign).